src/login/logind-dbus.c

```diff
@@ -1523,7 +1526,7 @@ static int bus_manager_log_shutdown(
                 p = strjoina(p, " (", m->wall_message, ").");
 
         return log_struct(LOG_NOTICE,
-                          "MESSAGE_ID=" SD_MESSAGE_SHUTDOWN_STR,
+                          streq(unit_name, SPECIAL_FACTORY_RESET_TARGET) ? "MESSAGE_ID=" SD_MESSAGE_FACTORY_RESET_STR : "MESSAGE_ID=" SD_MESSAGE_SHUTDOWN_STR,
```

Comment:
i'd prefer if you'd add another local var here, maybe "d" or so, which you set in the if blocks above, and just pass here, just like p + q.

Reply:
local var approach breaks tests(because log_struct second argument is not a string literal anymore). May revert to the ternary operator approach?

Reply:
reimplemented using separate function